src/Core/Types/Types/Relay/ConnectionMiddleware.cs

```diff
@@ -30,7 +30,11 @@ public ConnectionMiddleware(FieldDelegate next)
                 context.Argument<string>("after"),
                 context.Argument<string>("before"));
 
-            if (connectionResolver is { } && context.Result is TSource source)
+            if (context.Result is IConnection)
```

Comment:
I will have to look at the new implementation so that this will work with 11 as well.